### PR TITLE
Swap in real Rails endpoints & support SMS form submit

### PR DESCRIPTION
**Pull request by dstavis**
## Description:

…postman url in separate variable for future use. Remove commented-out code temporarily for readability.
Ready to review and merge.

The user can submit the SMS signup form, and sees a success message or an error message depending on the response from the server.

## Changes:

- The disaster URL has been swapped out
- successfully getting an empty array in response for my location
- displays alerts when the location has real alerts.

### CODE REVIEW

**Comment by dstavis** [comment]
Added User POST request. Getting 500 error, details in screenshots.
Screen Shot 2022-10-21 at 10 05 28 PM
Screen Shot 2022-10-21 at 10 04 19 PM

**Comment by dstavis** [comment]
Finished! 
Screen Shot 2022-10-24 at 3 37 53 PM
Screen Shot 2022-10-24 at 3 53 04 PM

**Comment by dstavis** [comment]
Please use this PR for code review @dara-rockwell @KatBrandt

File: src/components/App/App.js [hunk 1]
```diff
@@ -9,70 +9,84 @@ import SMSNotification from "../SMSNotification/SMSNotification";
...
 
 function App() {
 
   const [location, setLocation] = useState("")
+  const [smsFeedback, setSMSFeedback] = useState("")
   const [currentWeather, setCurrentWeather] = useState("")
   const [forecast, setForecast] = useState("")
   const [disasterAlerts, setDisasterAlerts] = useState("")
 
+  const formUserData = (phoneNumber) => {
```
Comment (nicktu12) [inline src/components/App/App.js:23]: Great SRP throughout your React code!

File: src/components/App/App.js [hunk 2]
```diff
@@ -9,70 +9,84 @@ import SMSNotification from "../SMSNotification/SMSNotification";
...
+    return {
+      "lat": String(location.lat),
+      "long": String(location.lng),
+      "phone": String(phoneNumber),
+    }
+  }
+
+  const addUser = (phoneNumber) => {
+    const userData = formUserData(phoneNumber)
+    fetch(ADD_USER_URL, {
```
Comment (nicktu12) [inline src/components/App/App.js:33]: There's an opportunity to move all API interactions to an utils file and clean up App.js

File: src/components/App/App.js [hunk 3]
```diff
@@ -9,70 +9,84 @@ import SMSNotification from "../SMSNotification/SMSNotification";
...
+      }
+    )
+    .catch( error => {
+      setSMSFeedback({ type: "error", message: `Error. Something went wrong. More info: ${error}`})
+    })
+  }
+
+  const formDisasterURLWithCoords = () => {
+    // Should we move this functionality into api.js?
+    return `${DISASTER_API_URL}?lat=${location.lat}&long=${location.lng}`
```
Comment (nicktu12) [inline src/components/App/App.js:56]: Cool!

File: src/components/App/App.js [hunk 4]
```diff
@@ -9,70 +9,84 @@ import SMSNotification from "../SMSNotification/SMSNotification";
...
+    }
+  }, [location])
 
   return (
     <div className="App">
       <main>
         <NavBar className='navbar'/>
         <div className="sms-search-weather">
-          <SMSNotification />
+          <SMSNotification addUser={addUser} userMessage={smsFeedback} />
```
Comment (nicktu12) [inline src/components/App/App.js:88]: Great use of data down, events up

File: src/components/SMSNotification/SMSNotification.css [hunk 5]
```diff
@@ -42,6 +43,28 @@
...
 
+.success {
+  color: green;
+  background-color: rgba(173, 255, 47, .3);
+  border-radius: 10px;
+  width: 90%;
+  padding: 10px;
+  margin-left: auto;
+  margin-right: auto;
+  margin-bottom: 10px;
```
Comment (nicktu12) [inline src/components/SMSNotification/SMSNotification.css:54]: These could be refactored with CSS below into a single rule for `.success, .error` to reduce duplicated code.

File: src/components/SMSNotification/SMSNotification.js [hunk 6]
```diff
@@ -1,24 +1,36 @@
...
-import 'react-phone-input-2/lib/style.css'
-import './SMSNotification.css'
+import React, { useState } from "react";
+import PhoneInput from "react-phone-input-2";
+import "react-phone-input-2/lib/style.css";
+import "./SMSNotification.css";
 
-const SMSNotification = () => {
-    const [ phoneNumber, setPhoneNumber ] = useState();
+const SMSNotification = (props) => {
```
Comment (nicktu12) [inline src/components/SMSNotification/SMSNotification.js:6]: Simple, very nice reusable component here.

File: src/components/SMSNotification/SMSNotification.css (hunk 5 above)
Comment (dstavis) [reply src/components/SMSNotification/SMSNotification.css:54]: Oh cool! I'll definitely do that.

File: src/api.js [hunk 7]
```diff
@@ -9,14 +9,16 @@ export const geoApiOptions = {
 export const GEO_API_URL = "https://wft-geo-db.p.rapidapi.com/v1/geo";
 export const WEATHER_API_URL = "https://api.openweathermap.org/data/2.5"
 export const WEATHER_API_KEY = "7454d65faa975f224f6bc598ccfe852f"
-// export const DISASTER_API_URL = "https://dc6e72c4-8622-4280-9089-79102851df02.mock.pstmn.io/api/v1/disasters?lat=33.2896&long=-97.6982" //MOCK DATA
 
-export const DISASTER_API_URL = "https://stay-or-go-server.herokuapp.com/api/v1"
-// export const DISASTER_API_URL = "https://stay-or-go-server.herokuapp.com/api/v1/disasters?lat=34.382692&long=-94.805750"
+export const DISASTER_API_URL = "https://stay-or-go-server.herokuapp.com/api/v1/disasters"
+export const DISASTER_MOCK_URL = "https://dc6e72c4-8622-4280-9089-79102851df02.mock.pstmn.io/api/v1/disasters?lat=33.2896&long=-97.6982"
```
Comment (dara-rockwell) [inline src/api.js:14]: ⛏️ This looks good! Are we sure we want to hard-code the `long` and `lat`?

File: src/api.js [hunk 8]
```diff
@@ -9,14 +9,16 @@ export const geoApiOptions = {
...
 
-// const fetchDisasterData = () => {
-// 	return fetch("https://dc6e72c4-8622-4280-9089-79102851df02.mock.pstmn.io/api/v1/disasters?lat=33.2896&long=-97.6982")
-// 	.then(response => response.json())
+export const ADD_USER_URL = "https://stay-or-go-server.herokuapp.com/api/v1/users"
+// required format for add user request body
+// {
```
Comment (dara-rockwell) [inline src/api.js:18]: ⛏️ Maybe remove commented code? We can always get it back from the commits.

File: src/api.js (hunk 7 above)
Comment (dara-rockwell) [reply src/api.js:14]: Now that I see how `DISASTER_API_URL` is used below, I think the only suggestion I might have is to move your mock URLs into a different file.

**Review by dara-rockwell — APPROVED** [review]

**Comment by dstavis** [comment]
I believe this branch's work has now been folded into https://github.com/capstone-stayOrGo/stayOrGo-client/pull/18
So we can close this PR without merging it

**Pull request closed without merging**